src/lib.rs

```diff
@@ -1,59 +1,120 @@
 #![allow(clippy::new_ret_no_self)]
+use evtx::{
+    EvtxParser, IntoIterChunks, JsonOutput, ParserSettings, SerializedEvtxRecord, XmlOutput,
+};
 
-use evtx::{EvtxParser, SerializedEvtxRecord, JsonOutput, IntoIterChunks, ParserSettings, XmlOutput};
-use pyo3::exceptions::{RuntimeError, NotImplementedError};
+use pyo3::exceptions::{NotImplementedError, RuntimeError};
 use pyo3::prelude::*;
-use pyo3::types::PyDict;
+use pyo3::types::{PyString};
+use pyo3::types::{PyDict};
+
 use pyo3::PyIterProtocol;
+use pyo3_file::PyFileLikeObject;
 
 use std::fs::File;
+use std::io;
+use std::io::{Read, Seek, SeekFrom};
+
+
+pub trait ReadSeek: Read + Seek {
+    fn tell(&mut self) -> io::Result<u64> {
+        self.seek(SeekFrom::Current(0))
+    }
+}
+
+impl<T: Read + Seek> ReadSeek for T {}
 
 struct PyEvtxError(evtx::err::Error);
 
 impl From<PyEvtxError> for PyErr {
     fn from(err: PyEvtxError) -> Self {
         match err.0 {
-            evtx::err::Error::IO { source, backtrace: _ } => {
-                source.into()
-            }
-            _ => {
-                PyErr::new::<RuntimeError, _>(format!("{}", err.0))
-            }
+            evtx::err::Error::IO {
+                source,
+                backtrace: _,
+            } => source.into(),
+            _ => PyErr::new::<RuntimeError, _>(format!("{}", err.0)),
         }
     }
 }
 
-
 #[derive(Copy, Clone, PartialOrd, PartialEq)]
 pub enum OutputFormat {
     JSON,
     XML,
 }
 
+#[derive(Debug)]
+enum FileOrFileLike {
+    File(String),
+    FileLike(PyFileLikeObject),
+}
+
+impl FileOrFileLike {
+    pub fn from_pyobject(path_or_file_like: PyObject) -> PyResult<FileOrFileLike> {
+        let gil = Python::acquire_gil();
+        let py = gil.python();
+
+        // is a path
+        if let Ok(string_ref) = path_or_file_like.cast_as::<PyString>(py) {
+            return Ok(FileOrFileLike::File(
+                string_ref.to_string_lossy().to_string(),
+            ));
+        }
+
+        // is a file-like
+        match PyFileLikeObject::new(path_or_file_like) {
```

Comment:
https://github.com/omerbenamram/pyo3-file/blob/master/src/lib.rs#L35
This can be a bit too strict because custom file-like stuff, like
![image](https://user-images.githubusercontent.com/2358365/58031704-3a136e80-7b21-11e9-8978-8ad4191f851a.png)

Reply:
Will now only validate read and seek.